fooof/Compute_broadbandpower_offset_slope.py

```diff
@@ -0,0 +1,347 @@
+# or a single roi (e.g. (10,) if you want to evaluate roi 11
+nr_rois = np.arange(78) # give list of rois or 1 rois (AAL cortical = 78, BNA cortical = 210)
+#nr_rois = (10,) # only run for roi 10 # note that python idexes at 0!
+
+nr_epochs= 10 # number of epochs
+Fs = 1250 # sample frequency
+freq_range=[0.5, 48] # frequency range you want to analyze
+
+subject_list = '/data/KNW/t.numan/GOALS/example_MEG_list.csv'
+dir_input = '/data/KNW/NO-cohorten/Scans/'
+dir_output = '/data/KNW/t.numan/GOALS/tmp/'
+save_output = True # you can save output
+name_to_save = 'test' # provide relevant name to save your run
+
+
+###########################
+# Run analysis            # 
+###########################
+
+# read the csv file containing Case_ID and MM of each included MEG
+# see example '/data/KNW/t.numan/GOALS/example_MEG_list.csv'
+subjects = pd.read_csv(subject_list, delimiter=';', header=0, usecols=[0,1])
+subjects['Case_ID'] = subjects['Case_ID'].apply(lambda x: str(x).zfill(4)) # add padding zeros when necessary
+
+# get location of ASCIIs of MEG 
+subjects['location_MEG'] = dir_input + 'sub-'+ subjects['Case_ID'] + '/meg/' + subjects['MM'] + '/' + atlas + '/' 
+
+
+###### Compute power spectrum ######
+## takes a while when multiple subjects are analyzed at once
+mean_pxx, broadband_power, f = run_loop_powerspectrum(subjects, nr_epochs=nr_epochs, nr_rois=nr_rois, 
+                                                      Fs=Fs, window_length=4096, freq_range=freq_range)
+       
+# mean_pxx contains the average power spectra over nr_epochs for all subjects for all rois
+# broadband_power is the area under the average power spectra over the frequency range for all subjects and rois
+# f gives the frequencies of the power spectrum (can be useful when plotting power spectrum)
+
+# save output
+if save_output == True:
+    np.save(dir_output + name_to_save + '_mean_pxx', mean_pxx)
+    np.save(dir_output + name_to_save + '_broadband_power', broadband_power)
+    print('saving power spectra and broadband power')
+    
+
+
+
+###### Compute FOOOOF offset and slope ######
+# create empty arrays to store offset and slope values
+offset = np.empty([mean_pxx.shape[0], mean_pxx.shape[2]]) # index 0 -> number of subjects, index 2 -> number of rois
+slope = np.empty([mean_pxx.shape[0], mean_pxx.shape[2]])
+
+print('Running FOOOF.........................')
+
+# run across all subjects in your list
+for sub in range(len(subjects)):
+    print('subject : ' + str(sub)) # print which subject is analyzed
+    if np.isnan(mean_pxx[sub,1,0]): # if there is no mean_pxx present for a subject, set offset/slope to nan
+        offset[sub,] = np.nan
+        slope[sub,] = np.nan
+    else:
+        i=0
+        for roi in nr_rois:
+            print('roi : ' + str(roi))
+            offset[sub,i], slope[sub,i] = cal_FOOOF_parameters(mean_pxx[sub,:,i], f, freq_range=[0.5, 48])
+            i=i+1
```

Comment:
have a look at https://www.python.org/dev/peps/pep-0008/   Other recommendations section.

Reply:
I'm not sure what you suggest here

Reply:
The idea is that in this section of PEP8 they explain these operator lines and how they should be i = i + 1 instead of what it is right now..

Reply:
Ah I didn't know what I should improve, but now I see it!